aries_cloudagent/protocols/issue_credential/v2_0/formats/vc_di/handler.py

```diff
@@ -0,0 +1,567 @@
+            },
+            "credential": {
+                "@context": [
+                    "https://www.w3.org/2018/credentials/v1",
+                    "https://w3id.org/security/data-integrity/v2",
+                    {"@vocab": "https://www.w3.org/ns/credentials/issuer-dependent#"},
+                ],
+                "type": ["VerifiableCredential"],
+                "issuer": public_did,
+                "credentialSubject": cred_proposal_message.credential_preview.attr_dict(),
+                "issuanceDate": datetime.datetime.now(
+                    datetime.timezone.utc
+                ).isoformat(),
+            },
+        }
+        return self.get_format_data(CRED_20_OFFER, vcdi_cred_offer)
+
+    async def receive_offer(
+        self, cred_ex_record: V20CredExRecord, cred_offer_message: V20CredOffer
+    ) -> None:
+        """Receive vcdi credential offer."""
+
+    async def create_request(
+        self, cred_ex_record: V20CredExRecord, request_data: Mapping = None
+    ) -> CredFormatAttachment:
+        """Create vcdi credential request."""
+        if cred_ex_record.state != V20CredExRecord.STATE_OFFER_RECEIVED:
+            raise V20CredFormatError(
+                "vcdi issue credential format cannot start from credential request"
+            )
+
+        await self._check_uniqueness(cred_ex_record.cred_ex_id)
+
+        holder_did = request_data.get("holder_did") if request_data else None
+        cred_offer = cred_ex_record.cred_offer.attachment(VCDICredFormatHandler.format)
+
+        if (
+            "anoncreds_link_secret" in cred_offer["binding_method"]
+            and "nonce" not in cred_offer["binding_method"]["anoncreds_link_secret"]
+        ):
+            raise V20CredFormatError(
+                "Missing nonce in credential offer with anoncreds link secret "
+                "binding method"
+            )
+
+        nonce = cred_offer["binding_method"]["anoncreds_link_secret"]["nonce"]
+        cred_def_id = cred_offer["binding_method"]["anoncreds_link_secret"][
+            "cred_def_id"
+        ]
+
+        ledger = self.profile.inject(BaseLedger)
+        multitenant_mgr = self.profile.inject_or(BaseMultitenantManager)
+        if multitenant_mgr:
+            ledger_exec_inst = IndyLedgerRequestsExecutor(self.profile)
+        else:
+            ledger_exec_inst = self.profile.inject(IndyLedgerRequestsExecutor)
+        ledger = (
+            await ledger_exec_inst.get_ledger_for_identifier(
+                cred_def_id,
+                txn_record_type=GET_CRED_DEF,
+            )
+        )[1]
+
+        async with ledger:
+            schema_id = await ledger.credential_definition_id2schema_id(cred_def_id)
+
+        async def _create():
+            anoncreds_registry = self.profile.inject(AnonCredsRegistry)
+
+            cred_def_result = await anoncreds_registry.get_credential_definition(
+                self.profile, cred_def_id
+            )
+
+            legacy_offer = {
```

Comment:
Same comment as above re converting between vcdi and "legacy" format for offer, request, etc